PK fails getting NPM dependency URL

```
Caused by: java.lang.NullPointerException: Cannot invoke "jakarta.json.JsonString.getString()" because the return value of "org.eclipse.parsson.JsonObjectBuilderImpl$JsonObjectImpl.getJsonString(String)" is null
    at org.eclipse.parsson.JsonObjectBuilderImpl$JsonObjectImpl.getString (JsonObjectBuilderImpl.java:264)
    at com.exasol.projectkeeper.sources.analyze.npm.NpmDependencies.getUrl (NpmDependencies.java:69)
    at com.exasol.projectkeeper.sources.analyze.npm.NpmDependencies.projectDependency (NpmDependencies.java:54)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:197)
    at java.util.ArrayList$ArrayListSpliterator.forEachRemaining (ArrayList.java:1625)
    at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:509)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto (AbstractPipeline.java:499)
    at java.util.stream.ReduceOps$ReduceOp.evaluateSequential (ReduceOps.java:921)
    at java.util.stream.AbstractPipeline.evaluate (AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.collect (ReferencePipeline.java:682)
    at com.exasol.projectkeeper.sources.analyze.npm.NpmDependencies.getDependencies (NpmDependencies.java:40)
    at com.exasol.projectkeeper.sources.analyze.npm.NpmServices.getDependencies (NpmServices.java:53)
    at com.exasol.projectkeeper.sources.analyze.npm.NpmSourceAnalyzer.analyzeSource (NpmSourceAnalyzer.java:52)
    at com.exasol.projectkeeper.sources.analyze.npm.NpmSourceAnalyzer.lambda$analyze$0 (NpmSourceAnalyzer.java:36)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:197)
    at java.util.ArrayList$ArrayListSpliterator.forEachRemaining (ArrayList.java:1625)
    at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:509)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto (AbstractPipeline.java:499)
    at java.util.stream.ReduceOps$ReduceOp.evaluateSequential (ReduceOps.java:921)
    at java.util.stream.AbstractPipeline.evaluate (AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.collect (ReferencePipeline.java:682)
    at com.exasol.projectkeeper.sources.analyze.npm.NpmSourceAnalyzer.analyze (NpmSourceAnalyzer.java:36)
    at com.exasol.projectkeeper.sources.SourceAnalyzer.analyzeSources (SourceAnalyzer.java:68)
    at com.exasol.projectkeeper.sources.SourceAnalyzer.analyze (SourceAnalyzer.java:59)
    at com.exasol.projectkeeper.ProjectKeeper.phase2AnalyzeSources (ProjectKeeper.java:137)
    at com.exasol.projectkeeper.ProjectKeeper.runValidationPhases (ProjectKeeper.java:276)
    at com.exasol.projectkeeper.ProjectKeeper.fix (ProjectKeeper.java:260)
    at com.exasol.projectkeeper.plugin.ProjectKeeperFixMojo.runProjectKeeper (ProjectKeeperFixMojo.java:20)
    at com.exasol.projectkeeper.plugin.AbstractProjectKeeperMojo.execute (AbstractProjectKeeperMojo.java:43)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
```

--- a/.github/workflows/dependencies_update.yml
+++ b/.github/workflows/dependencies_update.yml
@@ -68,14 +68,6 @@ jobs:
         env: {
           CREATED_ISSUES: '${{ inputs.vulnerability_issues }}'
         }
-      - name: Project Keeper Fix
-        id: project-keeper-fix
-        run: |
-          mvn --batch-mode com.exasol:project-keeper-maven-plugin:fix --projects .
-      - name: Project Keeper Fix for updated Project Keeper version
-        id: project-keeper-fix-2
-        run: |
-          mvn --batch-mode com.exasol:project-keeper-maven-plugin:fix --projects .
       - name: Generate Pull Request comment
         id: pr-comment
         run: |
@@ -88,7 +80,11 @@ jobs:
             echo 'It updates dependencies.' >> "$GITHUB_OUTPUT"
           fi
           echo >> "$GITHUB_OUTPUT"
-          echo '# ⚠️ This PR does not trigger CI workflows by default ⚠️' >> "$GITHUB_OUTPUT"
+          echo '# ⚠️ Notes ⚠️' >> "$GITHUB_OUTPUT"
+          echo '## Run PK fix manually' >> "$GITHUB_OUTPUT"
+          echo 'Due to restrictions workflow `dependencies_update.yml` can't update other workflows, see https://github.com/exasol/project-keeper/issues/578 for details.' >> "$GITHUB_OUTPUT"
+          echo 'Please checkout this PR locally and run `mvn com.exasol:project-keeper-maven-plugin:fix --projects .`' >> "$GITHUB_OUTPUT"
+          echo '## This PR does not trigger CI workflows' >> "$GITHUB_OUTPUT"
           echo 'Please click the **Close pull request** button and then **Reopen pull request** to trigger running checks.' >> "$GITHUB_OUTPUT"
           echo 'See https://github.com/exasol/project-keeper/issues/534 for details.' >> "$GITHUB_OUTPUT"
           echo 'EOF' >> "$GITHUB_OUTPUT"

--- a/dependencies.md
+++ b/dependencies.md
@@ -213,7 +213,7 @@
 | [org.sonatype.ossindex.maven:ossindex-maven-plugin][33] | [ASL2][34]                       |
 | [Maven Surefire Plugin][35]                             | [Apache-2.0][1]                  |
 | [Versions Maven Plugin][36]                             | [Apache License, Version 2.0][1] |
-| [Apache Maven JAR Plugin][62]                           | [Apache License, Version 2.0][1] |
+| [Apache Maven JAR Plugin][62]                           | [Apache-2.0][1]                  |
 | [duplicate-finder-maven-plugin Maven Mojo][37]          | [Apache License 2.0][38]         |
 | [Apache Maven Deploy Plugin][4]                         | [Apache-2.0][1]                  |
 | [Apache Maven GPG Plugin][39]                           | [Apache-2.0][1]                  |

--- a/dependencies.md
+++ b/dependencies.md
@@ -362,7 +362,7 @@
 [51]: https://github.com/vdurmont/semver4j
 [52]: https://bitbucket.org/snakeyaml/snakeyaml
 [53]: https://bitbucket.org/snakeyaml/snakeyaml-engine
-[54]: https://maven.apache.org/ref/3.9.6/maven-model/
+[54]: https://maven.apache.org/ref/3.9.7/maven-model/
 [55]: https://www.jcabi.com/jcabi-github
 [56]: https://www.jcabi.com/LICENSE.txt
 [57]: https://github.com/exasol/maven-project-version-getter/
@@ -376,8 +376,8 @@
 [65]: https://github.com/exasol/artifact-reference-checker-maven-plugin/
 [66]: https://github.com/exasol/artifact-reference-checker-maven-plugin/blob/main/LICENSE
 [67]: https://maven.apache.org/plugin-tools/maven-plugin-annotations
-[68]: https://maven.apache.org/ref/3.9.6/maven-plugin-api/
-[69]: https://maven.apache.org/ref/3.9.6/maven-core/
+[68]: https://maven.apache.org/ref/3.9.7/maven-plugin-api/
+[69]: https://maven.apache.org/ref/3.9.7/maven-core/
 [70]: https://www.eclemma.org/jacoco/index.html
 [71]: https://maven.apache.org/plugin-tools/maven-plugin-plugin
 [72]: https://maven.apache.org/plugins/maven-dependency-plugin/

--- a/doc/changes/changelog.md
+++ b/doc/changes/changelog.md
@@ -1,5 +1,6 @@
 # Changes
 
+* [4.3.2](changes_4.3.2.md)
 * [4.3.1](changes_4.3.1.md)
 * [4.3.0](changes_4.3.0.md)
 * [4.2.0](changes_4.2.0.md)

--- a/doc/changes/changes_4.3.2.md
+++ b/doc/changes/changes_4.3.2.md
@@ -9,6 +9,7 @@ This release fixes creation of the automatic dependency update Pull Request. Due
 ## Bugfixes
 
 * #578: Fixed automatic dependency update
+* #577: Fixed reading NPM dependencies with a missing project URL
 
 ## Dependency Updates
 

--- a/doc/changes/changes_4.3.2.md
+++ b/doc/changes/changes_4.3.2.md
@@ -12,6 +12,13 @@ This release fixes creation of the automatic dependency update Pull Request. Due
 
 ## Dependency Updates
 
+### Project Keeper Root Project
+
+#### Plugin Dependency Updates
+
+* Updated `org.apache.maven.plugins:maven-deploy-plugin:3.1.1` to `3.1.2`
+* Updated `org.apache.maven.plugins:maven-enforcer-plugin:3.4.1` to `3.5.0`
+
 ### Project Keeper Shared Model Classes
 
 #### Plugin Dependency Updates
@@ -55,6 +62,7 @@ This release fixes creation of the automatic dependency update Pull Request. Due
 
 #### Plugin Dependency Updates
 
+* Updated `org.apache.maven.plugins:maven-assembly-plugin:3.6.0` to `3.7.1`
 * Updated `org.apache.maven.plugins:maven-enforcer-plugin:3.4.1` to `3.5.0`
 * Updated `org.apache.maven.plugins:maven-javadoc-plugin:3.6.3` to `3.7.0`
 * Updated `org.sonarsource.scanner.maven:sonar-maven-plugin:3.11.0.3922` to `4.0.0.4121`
@@ -69,7 +77,9 @@ This release fixes creation of the automatic dependency update Pull Request. Due
 #### Plugin Dependency Updates
 
 * Updated `org.apache.maven.plugins:maven-enforcer-plugin:3.4.1` to `3.5.0`
+* Updated `org.apache.maven.plugins:maven-jar-plugin:3.3.0` to `3.4.1`
 * Updated `org.apache.maven.plugins:maven-javadoc-plugin:3.6.3` to `3.7.0`
+* Updated `org.apache.maven.plugins:maven-plugin-plugin:3.11.0` to `3.13.1`
 * Updated `org.sonarsource.scanner.maven:sonar-maven-plugin:3.11.0.3922` to `4.0.0.4121`
 * Updated `org.sonatype.plugins:nexus-staging-maven-plugin:1.6.13` to `1.7.0`
 
@@ -83,6 +93,7 @@ This release fixes creation of the automatic dependency update Pull Request. Due
 
 * Updated `org.apache.maven.plugins:maven-enforcer-plugin:3.4.1` to `3.5.0`
 * Updated `org.apache.maven.plugins:maven-javadoc-plugin:3.6.3` to `3.7.0`
+* Updated `org.apache.maven.plugins:maven-plugin-plugin:3.11.0` to `3.13.1`
 * Updated `org.sonarsource.scanner.maven:sonar-maven-plugin:3.11.0.3922` to `4.0.0.4121`
 * Updated `org.sonatype.plugins:nexus-staging-maven-plugin:1.6.13` to `1.7.0`
 

--- a/doc/changes/changes_4.3.2.md
+++ b/doc/changes/changes_4.3.2.md
@@ -1,0 +1,98 @@
+# Project Keeper 4.3.2, released 2024-06-03
+
+Code name: Fix automatic dependency update
+
+## Summary
+
+This release fixes creation of the automatic dependency update Pull Request. Due to restrictions of GitHub workflows it is not possible to run `project-keeper fix` which potentially updates other workflows. Please follow the instructions in the generated Pull Request and run PK fix manually. See #578 for details.
+
+## Bugfixes
+
+* #578: Fixed automatic dependency update
+
+## Dependency Updates
+
+### Project Keeper Shared Model Classes
+
+#### Plugin Dependency Updates
+
+* Updated `org.apache.maven.plugins:maven-enforcer-plugin:3.4.1` to `3.5.0`
+* Updated `org.apache.maven.plugins:maven-javadoc-plugin:3.6.3` to `3.7.0`
+* Updated `org.sonarsource.scanner.maven:sonar-maven-plugin:3.11.0.3922` to `4.0.0.4121`
+* Updated `org.sonatype.plugins:nexus-staging-maven-plugin:1.6.13` to `1.7.0`
+
+### Project Keeper Core
+
+#### Compile Dependency Updates
+
+* Updated `com.exasol:project-keeper-shared-model-classes:4.3.1` to `4.3.2`
+
+#### Runtime Dependency Updates
+
+* Updated `com.exasol:project-keeper-java-project-crawler:4.3.1` to `4.3.2`
+
+#### Test Dependency Updates
+
+* Updated `com.exasol:project-keeper-shared-test-setup:4.3.1` to `4.3.2`
+
+#### Plugin Dependency Updates
+
+* Updated `org.apache.maven.plugins:maven-enforcer-plugin:3.4.1` to `3.5.0`
+* Updated `org.apache.maven.plugins:maven-javadoc-plugin:3.6.3` to `3.7.0`
+* Updated `org.sonarsource.scanner.maven:sonar-maven-plugin:3.11.0.3922` to `4.0.0.4121`
+* Updated `org.sonatype.plugins:nexus-staging-maven-plugin:1.6.13` to `1.7.0`
+
+### Project Keeper Command Line Interface
+
+#### Compile Dependency Updates
+
+* Updated `com.exasol:project-keeper-core:4.3.1` to `4.3.2`
+* Updated `org.apache.maven:maven-model:3.9.6` to `3.9.7`
+
+#### Test Dependency Updates
+
+* Updated `com.exasol:project-keeper-shared-test-setup:4.3.1` to `4.3.2`
+
+#### Plugin Dependency Updates
+
+* Updated `org.apache.maven.plugins:maven-enforcer-plugin:3.4.1` to `3.5.0`
+* Updated `org.apache.maven.plugins:maven-javadoc-plugin:3.6.3` to `3.7.0`
+* Updated `org.sonarsource.scanner.maven:sonar-maven-plugin:3.11.0.3922` to `4.0.0.4121`
+* Updated `org.sonatype.plugins:nexus-staging-maven-plugin:1.6.13` to `1.7.0`
+
+### Project Keeper Maven Plugin
+
+#### Compile Dependency Updates
+
+* Updated `com.exasol:project-keeper-core:4.3.1` to `4.3.2`
+
+#### Plugin Dependency Updates
+
+* Updated `org.apache.maven.plugins:maven-enforcer-plugin:3.4.1` to `3.5.0`
+* Updated `org.apache.maven.plugins:maven-javadoc-plugin:3.6.3` to `3.7.0`
+* Updated `org.sonarsource.scanner.maven:sonar-maven-plugin:3.11.0.3922` to `4.0.0.4121`
+* Updated `org.sonatype.plugins:nexus-staging-maven-plugin:1.6.13` to `1.7.0`
+
+### Project Keeper Java Project Crawler
+
+#### Compile Dependency Updates
+
+* Updated `com.exasol:project-keeper-shared-model-classes:4.3.1` to `4.3.2`
+
+#### Plugin Dependency Updates
+
+* Updated `org.apache.maven.plugins:maven-enforcer-plugin:3.4.1` to `3.5.0`
+* Updated `org.apache.maven.plugins:maven-javadoc-plugin:3.6.3` to `3.7.0`
+* Updated `org.sonarsource.scanner.maven:sonar-maven-plugin:3.11.0.3922` to `4.0.0.4121`
+* Updated `org.sonatype.plugins:nexus-staging-maven-plugin:1.6.13` to `1.7.0`
+
+### Project Keeper Shared Test Setup
+
+#### Compile Dependency Updates
+
+* Updated `com.exasol:project-keeper-shared-model-classes:4.3.1` to `4.3.2`
+
+#### Plugin Dependency Updates
+
+* Updated `org.apache.maven.plugins:maven-enforcer-plugin:3.4.1` to `3.5.0`
+* Updated `org.sonarsource.scanner.maven:sonar-maven-plugin:3.11.0.3922` to `4.0.0.4121`

--- a/doc/requirements/design.md
+++ b/doc/requirements/design.md
@@ -745,7 +745,7 @@ Covers:
 * [`dsn~dependency-updater.workflow.generate~1`](#generate-dependencies_updateyml-workflow)
 
 #### `dependencies_update.yml` Workflow Starts PK `fix` Mode Twice
-`dsn~dependency-updater.workflow.start-pk-fix~1`
+`dsn~dependency-updater.workflow.start-pk-fix~2`
 
 PK generates the `dependencies_update.yml` workflow so that it starts PK's `fix` mode twice after updating dependencies.
 

--- a/doc/requirements/design.md
+++ b/doc/requirements/design.md
@@ -754,7 +754,10 @@ Rationale:
 * After updating dependency versions, the workflow runs PK `fix` to update the dependencies section in the changes file.
 * PK `fix` will potentially update PK to a newer version in `pom.xml` (see [`dsn~self-update~1`](#project-keeper-self-update)). So the workflow runs PK `fix` a second time (using the updated PK version) to update all generated files.
 
-Needs: impl
+Note:
+Due to restrictions of GitHub permissions it is not possible for a workflow triggered by an event to modify GitHub workflows (see https://github.com/orgs/community/discussions/35410). That's why pushing the changes fail. A workaround would be to use a separate PAT, but this is too much effort.
+
+We decided to skip running PK fix. The user must checkout the branch locally and run PK fix manually.
 
 Covers:
 * [`dsn~dependency-updater.workflow.generate~1`](#generate-dependencies_updateyml-workflow)

--- a/maven-project-crawler/pk_generated_parent.pom
+++ b/maven-project-crawler/pk_generated_parent.pom
@@ -64,7 +64,7 @@
             <plugin>
                 <groupId>org.sonarsource.scanner.maven</groupId>
                 <artifactId>sonar-maven-plugin</artifactId>
-                <version>3.11.0.3922</version>
+                <version>4.0.0.4121</version>
             </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
@@ -102,7 +102,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-enforcer-plugin</artifactId>
-                <version>3.4.1</version>
+                <version>3.5.0</version>
                 <executions>
                     <execution>
                         <id>enforce-maven</id>
@@ -273,7 +273,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
-                <version>3.6.3</version>
+                <version>3.7.0</version>
                 <executions>
                     <execution>
                         <id>attach-javadocs</id>
@@ -295,7 +295,7 @@
             <plugin>
                 <groupId>org.sonatype.plugins</groupId>
                 <artifactId>nexus-staging-maven-plugin</artifactId>
-                <version>1.6.13</version>
+                <version>1.7.0</version>
                 <configuration>
                     <autoReleaseAfterClose>true</autoReleaseAfterClose>
                     <serverId>ossrh</serverId>

--- a/parent-pom/pom.xml
+++ b/parent-pom/pom.xml
@@ -259,7 +259,7 @@
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-plugin-plugin</artifactId>
-                    <version>3.11.0</version>
+                    <version>3.13.1</version>
                 </plugin>
             </plugins>
         </pluginManagement>

--- a/parent-pom/pom.xml
+++ b/parent-pom/pom.xml
@@ -28,10 +28,10 @@
         </repository>
     </distributionManagement>
     <properties>
-        <revision>4.3.1</revision>
+        <revision>4.3.2</revision>
         <!-- Integration test ProjectKeeperMojoIT starts a Maven build which requires Java 17. -->
         <java.version>17</java.version>
-        <maven.version>3.9.6</maven.version>
+        <maven.version>3.9.7</maven.version>
         <minimum.maven.version>3.6.3</minimum.maven.version>
         <junit.version>5.10.2</junit.version>
         <xmlunit.version>2.10.0</xmlunit.version>
@@ -89,7 +89,7 @@
             <dependency>
                 <groupId>org.apache.maven.plugin-tools</groupId>
                 <artifactId>maven-plugin-annotations</artifactId>
-                <version>3.13.0</version>
+                <version>3.13.1</version>
                 <scope>provided</scope>
             </dependency>
             <dependency>

--- a/pom.xml
+++ b/pom.xml
@@ -31,7 +31,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-enforcer-plugin</artifactId>
-                <version>3.4.1</version>
+                <version>3.5.0</version>
                 <executions>
                     <execution>
                         <id>enforce-maven</id>
@@ -70,7 +70,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-deploy-plugin</artifactId>
-                <version>3.1.1</version>
+                <version>3.1.2</version>
                 <configuration>
                     <skip>true</skip>
                 </configuration>

--- a/project-keeper-cli/pk_generated_parent.pom
+++ b/project-keeper-cli/pk_generated_parent.pom
@@ -56,7 +56,7 @@
             <plugin>
                 <groupId>org.sonarsource.scanner.maven</groupId>
                 <artifactId>sonar-maven-plugin</artifactId>
-                <version>3.11.0.3922</version>
+                <version>4.0.0.4121</version>
             </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
@@ -94,7 +94,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-enforcer-plugin</artifactId>
-                <version>3.4.1</version>
+                <version>3.5.0</version>
                 <executions>
                     <execution>
                         <id>enforce-maven</id>
@@ -315,7 +315,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
-                <version>3.6.3</version>
+                <version>3.7.0</version>
                 <executions>
                     <execution>
                         <id>attach-javadocs</id>
@@ -337,7 +337,7 @@
             <plugin>
                 <groupId>org.sonatype.plugins</groupId>
                 <artifactId>nexus-staging-maven-plugin</artifactId>
-                <version>1.6.13</version>
+                <version>1.7.0</version>
                 <configuration>
                     <autoReleaseAfterClose>true</autoReleaseAfterClose>
                     <serverId>ossrh</serverId>

--- a/project-keeper-cli/pom.xml
+++ b/project-keeper-cli/pom.xml
@@ -75,7 +75,6 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-assembly-plugin</artifactId>
-                <version>3.6.0</version>
                 <configuration>
                     <finalName>project-keeper-cli-${project.version}</finalName>
                     <archive>

--- a/project-keeper-cli/pom.xml
+++ b/project-keeper-cli/pom.xml
@@ -98,17 +98,10 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
                 <configuration>
-                    <!-- Javadoc causes error "error: package java.util.logging is not visible
-                        import java.util.logging.*;
-                        (package java.util.logging is declared in module java.logging, but module jakarta.json does not read it)". -->
-                    <additionalOptions>
-                        <option>--add-modules</option>
-                        <option>java.logging</option>
-                        <option>--add-reads</option>
-                        <option>jakarta.json=java.logging</option>
-                    </additionalOptions>
-                    <!-- Ignore warning "warning: The code being documented uses modules but the packages defined in https://github.com/exasol/project-keeper/apidocs/ are in the unnamed module." with Java 17-->
-                    <failOnWarnings>false</failOnWarnings>
+                    <!-- Javadoc generation fails with error 'error: module not found: jakarta.xml.bind'.
+                        This option deactivates modules as we don't use modules anyway.
+                        See https://maven.apache.org/plugins/maven-javadoc-plugin/javadoc-mojo.html#legacymode -->
+                    <legacyMode>true</legacyMode>
                 </configuration>
             </plugin>
         </plugins>

--- a/project-keeper-maven-plugin/pk_generated_parent.pom
+++ b/project-keeper-maven-plugin/pk_generated_parent.pom
@@ -64,7 +64,7 @@
             <plugin>
                 <groupId>org.sonarsource.scanner.maven</groupId>
                 <artifactId>sonar-maven-plugin</artifactId>
-                <version>3.11.0.3922</version>
+                <version>4.0.0.4121</version>
             </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
@@ -102,7 +102,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-enforcer-plugin</artifactId>
-                <version>3.4.1</version>
+                <version>3.5.0</version>
                 <executions>
                     <execution>
                         <id>enforce-maven</id>
@@ -273,7 +273,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
-                <version>3.6.3</version>
+                <version>3.7.0</version>
                 <executions>
                     <execution>
                         <id>attach-javadocs</id>
@@ -295,7 +295,7 @@
             <plugin>
                 <groupId>org.sonatype.plugins</groupId>
                 <artifactId>nexus-staging-maven-plugin</artifactId>
-                <version>1.6.13</version>
+                <version>1.7.0</version>
                 <configuration>
                     <autoReleaseAfterClose>true</autoReleaseAfterClose>
                     <serverId>ossrh</serverId>

--- a/project-keeper-maven-plugin/pom.xml
+++ b/project-keeper-maven-plugin/pom.xml
@@ -108,7 +108,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-jar-plugin</artifactId>
-                <version>3.3.0</version>
+                <version>3.4.1</version>
                 <configuration>
                     <archive>
                         <manifest>

--- a/project-keeper/pk_generated_parent.pom
+++ b/project-keeper/pk_generated_parent.pom
@@ -56,7 +56,7 @@
             <plugin>
                 <groupId>org.sonarsource.scanner.maven</groupId>
                 <artifactId>sonar-maven-plugin</artifactId>
-                <version>3.11.0.3922</version>
+                <version>4.0.0.4121</version>
             </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
@@ -94,7 +94,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-enforcer-plugin</artifactId>
-                <version>3.4.1</version>
+                <version>3.5.0</version>
                 <executions>
                     <execution>
                         <id>enforce-maven</id>
@@ -265,7 +265,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
-                <version>3.6.3</version>
+                <version>3.7.0</version>
                 <executions>
                     <execution>
                         <id>attach-javadocs</id>
@@ -287,7 +287,7 @@
             <plugin>
                 <groupId>org.sonatype.plugins</groupId>
                 <artifactId>nexus-staging-maven-plugin</artifactId>
-                <version>1.6.13</version>
+                <version>1.7.0</version>
                 <configuration>
                     <autoReleaseAfterClose>true</autoReleaseAfterClose>
                     <serverId>ossrh</serverId>

--- a/project-keeper/src/main/java/com/exasol/projectkeeper/sources/analyze/npm/NpmDependencies.java
+++ b/project-keeper/src/main/java/com/exasol/projectkeeper/sources/analyze/npm/NpmDependencies.java
@@ -66,6 +66,9 @@ class NpmDependencies {
     }
 
     private String getUrl(final String moduleName) {
-        return this.additionalInfo.getJsonObject(moduleName).getString("resolved");
+        return Optional.ofNullable(this.additionalInfo.getJsonObject(moduleName)) //
+                .filter(object -> object.containsKey("resolved")) //
+                .map(object -> object.getString("resolved")) //
+                .orElse(null);
     }
 }

--- a/project-keeper/src/main/resources/maven_templates/maven-enforcer-plugin.xml
+++ b/project-keeper/src/main/resources/maven_templates/maven-enforcer-plugin.xml
@@ -1,7 +1,7 @@
 <plugin>
     <groupId>org.apache.maven.plugins</groupId>
     <artifactId>maven-enforcer-plugin</artifactId>
-    <version>3.4.1</version>
+    <version>3.5.0</version>
     <executions>
         <execution>
             <id>enforce-maven</id>

--- a/project-keeper/src/main/resources/maven_templates/maven-javadoc-plugin.xml
+++ b/project-keeper/src/main/resources/maven_templates/maven-javadoc-plugin.xml
@@ -1,7 +1,7 @@
 <plugin>
     <groupId>org.apache.maven.plugins</groupId>
     <artifactId>maven-javadoc-plugin</artifactId>
-    <version>3.6.3</version>
+    <version>3.7.0</version>
     <executions>
         <execution>
             <id>attach-javadocs</id>

--- a/project-keeper/src/main/resources/maven_templates/nexus-staging-maven-plugin.xml
+++ b/project-keeper/src/main/resources/maven_templates/nexus-staging-maven-plugin.xml
@@ -1,7 +1,7 @@
 <plugin>
     <groupId>org.sonatype.plugins</groupId>
     <artifactId>nexus-staging-maven-plugin</artifactId>
-    <version>1.6.13</version>
+    <version>1.7.0</version>
     <configuration>
         <autoReleaseAfterClose>true</autoReleaseAfterClose>
         <serverId>ossrh</serverId>

--- a/project-keeper/src/main/resources/maven_templates/sonar-maven-plugin.xml
+++ b/project-keeper/src/main/resources/maven_templates/sonar-maven-plugin.xml
@@ -1,5 +1,5 @@
 <plugin>
     <groupId>org.sonarsource.scanner.maven</groupId>
     <artifactId>sonar-maven-plugin</artifactId>
-    <version>3.11.0.3922</version>
+    <version>4.0.0.4121</version>
 </plugin>

--- a/project-keeper/src/main/resources/templates/.github/workflows/dependencies_update.yml
+++ b/project-keeper/src/main/resources/templates/.github/workflows/dependencies_update.yml
@@ -62,19 +62,6 @@ jobs:
         env:
           CREATED_ISSUES: ${{ inputs.vulnerability_issues }}
 
-      - name: Project Keeper Fix
-        id: project-keeper-fix
-        run: |
-          mvn --batch-mode com.exasol:project-keeper-maven-plugin:fix --projects .
-
-      - name: Project Keeper Fix for updated Project Keeper version
-        id: project-keeper-fix-2
-        # Calling PK fix a second time is necessary because the first invocation potentially updated PK itself.
-        # So we need to run PK fix again with the latest PK version.
-        # [impl->dsn~dependency-updater.workflow.start-pk-fix~1]
-        run: |
-          mvn --batch-mode com.exasol:project-keeper-maven-plugin:fix --projects .
-
       - name: Generate Pull Request comment
         id: pr-comment
         # [impl->dsn~dependency-updater.workflow.create-pull-request~1]
@@ -89,7 +76,11 @@ jobs:
             echo 'It updates dependencies.' >> "$GITHUB_OUTPUT"
           fi
           echo >> "$GITHUB_OUTPUT"
-          echo '# ⚠️ This PR does not trigger CI workflows by default ⚠️' >> "$GITHUB_OUTPUT"
+          echo '# ⚠️ Notes ⚠️' >> "$GITHUB_OUTPUT"
+          echo '## Run PK fix manually' >> "$GITHUB_OUTPUT"
+          echo 'Due to restrictions workflow `dependencies_update.yml` can't update other workflows, see https://github.com/exasol/project-keeper/issues/578 for details.' >> "$GITHUB_OUTPUT"
+          echo 'Please checkout this PR locally and run `mvn com.exasol:project-keeper-maven-plugin:fix --projects .`' >> "$GITHUB_OUTPUT"
+          echo '## This PR does not trigger CI workflows' >> "$GITHUB_OUTPUT"
           echo 'Please click the **Close pull request** button and then **Reopen pull request** to trigger running checks.' >> "$GITHUB_OUTPUT"
           echo 'See https://github.com/exasol/project-keeper/issues/534 for details.' >> "$GITHUB_OUTPUT"
           echo 'EOF' >> "$GITHUB_OUTPUT"

--- a/project-keeper/src/test/java/com/exasol/projectkeeper/sources/analyze/npm/NpmDependenciesTest.java
+++ b/project-keeper/src/test/java/com/exasol/projectkeeper/sources/analyze/npm/NpmDependenciesTest.java
@@ -39,6 +39,25 @@ class NpmDependenciesTest {
         ));
     }
 
+    @Test
+    void getDependenciesMissingUrl() {
+        when(this.npmServices.listDependencies(any())).thenReturn(json(TestData.DEPENDENCIES_MISSING_URL));
+        when(this.npmServices.getLicenses(any())).thenReturn(json(TestData.LICENSES));
+
+        final NpmDependencies testee = new NpmDependencies(this.npmServices, TestData.samplePackageJson());
+        assertThat(testee.getDependencies(), containsInAnyOrder( //
+                dependency(Type.DEV, "changed-plugin", "1.2.0", "CP"), //
+                dependency(Type.DEV, "new-plugin", "1.3.0", "NP"), //
+                dependency(Type.COMPILE, "changed-compile", "2.2.0", "CC"), //
+                ProjectDependency.builder() //
+                        .type(Type.COMPILE)//
+                        .name("new-compile") //
+                        .websiteUrl(null) //
+                        .licenses(List.of(new License("NC" + "-License", "https://" + "new-compile"))) //
+                        .build() //
+        ));
+    }
+
     private ProjectDependency dependency(final Type type, final String name, final String version,
             final String license) {
         return ProjectDependency.builder() //

--- a/project-keeper/src/test/java/com/exasol/projectkeeper/sources/analyze/npm/TestData.java
+++ b/project-keeper/src/test/java/com/exasol/projectkeeper/sources/analyze/npm/TestData.java
@@ -77,6 +77,28 @@ class TestData {
             "  }", //
             "}");
 
+    static final String DEPENDENCIES_MISSING_URL = jsonString("{", //
+            "  'name': 'module-name',", //
+            "  'version': '2.0.0',", //
+            "  'dependencies': {", //
+            "    'changed-plugin': {", //
+            "      'version': '1.2.0',", //
+            "      'resolved': 'https://changed-plugin/-/1.2.0.tgz'", //
+            "    },", //
+            "    'new-plugin': {", //
+            "      'version': '1.3.0',", //
+            "      'resolved': 'https://new-plugin/-/1.3.0.tgz'", //
+            "    },", //
+            "    'changed-compile': {", //
+            "      'version': '2.2.0',", //
+            "      'resolved': 'https://changed-compile/-/2.2.0.tgz'", //
+            "    },", //
+            "    'new-compile': {", //
+            "      'version': '2.3.0'", //
+            "    }", //
+            "  }", //
+            "}");
+
     static String jsonString(final String... s) {
         return String.join("\n", s).replace('\'', '"');
     }

--- a/project-keeper/src/test/java/com/exasol/projectkeeper/validators/files/CiBuildWorkflowGeneratorTest.java
+++ b/project-keeper/src/test/java/com/exasol/projectkeeper/validators/files/CiBuildWorkflowGeneratorTest.java
@@ -214,7 +214,7 @@ class CiBuildWorkflowGeneratorTest {
                                 .step(WorkflowStep.createStep(Map.of("id", "new-step", "name", "New Step"))).build())
                         .build()))
                 .build()).getJob("update_dependencies");
-        assertAll(() -> assertThat(job.getSteps(), hasSize(16)),
+        assertAll(() -> assertThat(job.getSteps(), hasSize(14)),
                 () -> assertThat(job.getStep("new-step").getName(), equalTo("New Step")));
     }
 
@@ -249,7 +249,7 @@ class CiBuildWorkflowGeneratorTest {
     @Test
     void dependencyUpdateBuildAllStepsHaveId() {
         final Job job = dependencyUpdateBuildContent(BuildOptions.builder().build()).getJob("update_dependencies");
-        assertThat(job.getSteps(), hasSize(greaterThanOrEqualTo(15)));
+        assertThat(job.getSteps(), hasSize(greaterThanOrEqualTo(13)));
         job.getSteps().forEach(step -> assertThat(step.getId(), notNullValue()));
     }
 

--- a/shared-model-classes/pk_generated_parent.pom
+++ b/shared-model-classes/pk_generated_parent.pom
@@ -56,7 +56,7 @@
             <plugin>
                 <groupId>org.sonarsource.scanner.maven</groupId>
                 <artifactId>sonar-maven-plugin</artifactId>
-                <version>3.11.0.3922</version>
+                <version>4.0.0.4121</version>
             </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
@@ -94,7 +94,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-enforcer-plugin</artifactId>
-                <version>3.4.1</version>
+                <version>3.5.0</version>
                 <executions>
                     <execution>
                         <id>enforce-maven</id>
@@ -265,7 +265,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
-                <version>3.6.3</version>
+                <version>3.7.0</version>
                 <executions>
                     <execution>
                         <id>attach-javadocs</id>
@@ -287,7 +287,7 @@
             <plugin>
                 <groupId>org.sonatype.plugins</groupId>
                 <artifactId>nexus-staging-maven-plugin</artifactId>
-                <version>1.6.13</version>
+                <version>1.7.0</version>
                 <configuration>
                     <autoReleaseAfterClose>true</autoReleaseAfterClose>
                     <serverId>ossrh</serverId>

--- a/shared-test-setup/pk_generated_parent.pom
+++ b/shared-test-setup/pk_generated_parent.pom
@@ -45,7 +45,7 @@
             <plugin>
                 <groupId>org.sonarsource.scanner.maven</groupId>
                 <artifactId>sonar-maven-plugin</artifactId>
-                <version>3.11.0.3922</version>
+                <version>4.0.0.4121</version>
             </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
@@ -83,7 +83,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-enforcer-plugin</artifactId>
-                <version>3.4.1</version>
+                <version>3.5.0</version>
                 <executions>
                     <execution>
                         <id>enforce-maven</id>

